ZMQ: fixed warnings

lmesz opened the pull request:
Hali,

Pleases review the warning fixes.

Thanks

juhaszviktor commented on modules/zmq/zmq-transport.c:
```diff
@@ -31,10 +31,10 @@ typedef struct _LogTransportZMQ
+log_transport_zmq_read_method(LogTransport *s, gpointer buf, gsize buflen, LogTransportAuxData *aux)
 {
   LogTransportZMQ *self = (LogTransportZMQ *) s;
-  return zmq_recv(self->socket, buf, buflen, ZMQ_DONTWAIT);
```
Why the space? It is not a compile warning correction, is it?

juhaszviktor commented:
Thank you very much for your contribution!

There is still only one compile warning. (zmq-grammar
See the travis log.
zmq-grammar.y:379:37: warning: assignment makes pointer from integer without a cast [enabled by default]

Thanks for your help :)

lmesz commented:
Hali,

Yes it was right about the recv(). And fixed some memory leaks. Please check them.

czanik commented:
I did a compile using the two patches, but I still receive this error in the post build checks:

[   26s] E: syslog-ng-incubator 64bit-portability-issue modules/zmq/zmq-grammar.y:379

Looking at the build logs:

[   22s]   CC       modules/zmq/modules_zmq_libzmq_la-zmq-grammar.lo
[   22s] modules/zmq/zmq-grammar.y: In function 'zmq_parse':
[   22s] modules/zmq/zmq-grammar.y:379:13: warning: implicit declaration of function 'zmq_sd_new' [-Wimplicit-function-declaration]
[   22s]              last_driver = *instance = zmq_sd_new(configuration);
[   22s]              ^
[   22s] modules/zmq/zmq-grammar.y:379:37: warning: assignment makes pointer from integer without a cast [enabled by default]
[   22s]              last_driver = *instance = zmq_sd_new(configuration);
[   22s]                                      ^

lmesz commented:
Upsz okay it is still not complete I will fix it also

lmesz committed:
zmq-source: change the way how zmq-source handle the connection creation

 * In the source case syslog-ng use pull socket. When you use pull socket you must connect to socket instead of bind and the other part has to bind.
 * E.g.: pull -> connect | push -> bind
 * Remove the lot of c test scripts and use python instead.

lmesz committed:
zmq: removed unused zmq-source and destination header files

lmesz committed:
zmq: added missing "dependency" it contains the init_sequence_number function

lmesz committed:
zmq: added minimal structure for testing at least to test if it can be loaded or not

lmesz committed:
zmq: added unit test to some zmq-source function and fix a bug around create_zmq_context

 * Bug: errno wasn't set to 0 before the creation of zmq socket. So if failed to create first zmq-socket the second followed it because the error variable contained the first socket error number.
 * function test script: changed default port from 8888 to 5558 to be symetric with the source code

lmesz committed:
zmq: made source function consistent every function starts with zmq_sd_* and followed the changes in test and module header

lmesz committed:
zmq: set socket type from destination
 * fixed a leak around insert, syslog-ng didn't free the result if the insertion to zmq_queue() was successful
 * made the function names consistent

lmesz committed:
zmq: refactored zmq-destination address -> host

lmesz committed:
zmq: refactor source grammar

lmesz committed:
zmq: added test case to destination

lmesz committed:
zmq: close socket when something goes wrong

lmesz committed:
zmq(WIP): added functional test

lmesz committed:
zmq: fixed errno value when everything goes wrong, do not return EAGAIN

lmesz committed:
zmq: changed zmq_ctx_term to destroy, it will be depreicated and need to changed to term, but now the packaged libzmq doesn not support this function

lmesz committed:
zmq: added queue to source

 * ZMQ contains flow-control itself
 * The queue is necessary to set path_options to not bother about
   acknowledgement
 * Previously there was a bug which ended in a segfault. After the first
   reload and message sending syslog-ng crashed because the bookmark
   wasn't set properly.

lmesz committed:
zmq: fixed path of runtesth.sh

lmesz committed:
zmq: remove term like before one remained. Forget to grep!

lmesz committed:
zmq(wip): use syslog-ng test infrastructure instead of own for func tests

lmesz committed:
zmq(wip): extended travis config

lmesz committed:
zmq(wip): one little step toward zmq-func-test

lmesz committed:
zmq: fixed plugin information

lmesz committed:
zmq(wip): check if SYSLOG_NG_BINARY set and if not use the system-wide used syslog-ng

lmesz committed:
zmq: waaa func-test

mochrul commented:
@lmesz please rebase your patch. Currently, this PR has conflicting files and therefore cannot be merged.
